### PR TITLE
Unreal engine Documentation update

### PR DESCRIPTION
**Pull request by dirtycajunrice**
Update all documentation to match with extra detail, and added kree for blueprint visualization of nodes.

---

## PR-Codex overview
The focus of this PR is to enhance the Unreal Engine SDK in the portal app by adding new functionalities, updating metadata, and restructuring the sidebar navigation.

### Detailed summary
- Added `@dirtycajunrice/klee` dependency
- Updated `Unreal` to `Unreal Engine` in navigation
- Added new functionalities for private key and in-app wallets
- Updated metadata titles and descriptions
- Restructured sidebar navigation for better organization

> The following files were skipped due to too many changes: `apps/portal/src/app/unreal/blueprints/utilities/page.mdx`, `apps/portal/src/app/unreal/blueprints/smart-wallet/page.mdx`, `apps/portal/src/app/unreal/cpp/common/page.mdx`, `apps/portal/src/app/unreal/getting-started/page.mdx`, `apps/portal/src/app/unreal/thirdweb-wallet-handle/page.mdx`, `apps/portal/src/app/unreal/page.mdx`, `apps/portal/src/app/unreal/thirdweb/page.mdx`, `apps/portal/src/app/unreal/blueprints/private-key-wallet/nodes.ts`, `legacy_packages/contracts-js/package.json`, `apps/portal/src/app/unreal/blueprints/utilities/nodes.ts`, `apps/portal/src/app/unreal/blueprints/in-app-wallet/nodes.ts`, `pnpm-lock.yaml`, `apps/portal/src/app/unreal/blueprints/smart-wallet/nodes.ts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Review by 0xFirekeeper — APPROVED** [review]

**Comment by dirtycajunrice** [comment]
Worth noting that this pr includes the changes from https://github.com/thirdweb-dev/js/pull/4385

File: apps/dashboard/package.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
   "private": true,
   "scripts": {
     "preinstall": "npx only-allow pnpm",
-    "dev:no-cache": "rm -rf .next/cache && pnpm run dev",
+    "dev:no-cache": "git clean -xdf && pnpm run dev",
```
Comment (joaquim-verges) [inline apps/dashboard/package.json:7]: doesnt that remove all the untracked files? like even my .envs and stuff?

**Comment by dirtycajunrice** [comment]
* **#4427** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4427?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @dirtycajunrice and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/dashboard/package.json (hunk 1 above)
Comment (dirtycajunrice) [reply apps/dashboard/package.json:7]: This was removed in the upstream change, but it only removes untracked files that are not in .gitignore so shouldnt touch .env

File: apps/portal/scripts/createEmptySearchIndex.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import { existsSync, writeFileSync } from "node:fs";
+import { existsSync, writeFileSync } from "fs";
```
Comment (jnsdls) [inline apps/portal/scripts/createEmptySearchIndex.ts:1]: change this back

File: apps/portal/scripts/extractSearchData.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import { writeFileSync } from "node:fs";
+import { writeFileSync } from "fs";
```
Comment (jnsdls) [inline apps/portal/scripts/extractSearchData.ts:1]: change this back

File: apps/portal/src/app/unreal/thirdweb-common/page.mdx
Comment (jnsdls) [inline apps/portal/src/app/unreal/thirdweb-common/page.mdx:1]: add redirects for the deleted paths

**Review by jnsdls — CHANGES_REQUESTED** [review]

**Review by jnsdls — APPROVED** [review]

**Pull request merged**